packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -47,21 +47,25 @@ contract BridgePool is Testable {
         address l1Token;
         address slowRelayer;
     }
+
+    // TODO: Rename Deposit --> Relay since the Relay is derived from the Deposit.
     struct Deposit {
         DepositState depositState;
         DepositType depositType;
         // A deposit can have both a slow and an instant relayer if a slow relay is "sped up" from slow to instant.
         // We want to store both addresses for separate payouts.
         address instantRelayer;
-        // @dev: See note above about why some Deposit params are collapsed into `RelayAncillaryDataContents`.
-        RelayAncillaryDataContents relayData;
+        // Store unique hash derived from all of the data neccessary to construct this deposit.
+        bytes32 relayHash;
     }
-    // Associates each deposit with a unique ID.
-    mapping(uint64 => Deposit) public deposits;
+    // Associates each deposit with a unique ancillary data derived from its constituent data. We need to key by the
+    // ancillary data so that the OptimisticOracle can locate deposits on callbacks using only price requests' ancillary
+    // data
+    mapping(bytes => Deposit) public deposits;
```

Comment:
It's effectively the same thing, but why not just use the relayHash directly as the key (bytes32), which would mean we don't need to reconstruct the hash to do the lookup.

Reply:
because of the `OptimisticOracle.priceDisputed()` [callback](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/implementation/OptimisticOracle.sol#L411). When a relay price request is disputed, we'd like for the OO to be able to callback into this method and move the pending relay to the pending dispute mapping but this callback is only aware of the price request's ancillary data, not the relay hash in this context.

Reply:
I actually changed this logic a bit to remove all of the hashing logic. Since we're key-ing by ancillary data, we only need to pass the ancillary data around.